tests/test_attributes.py

```diff
@@ -0,0 +1,257 @@
+                {% load component_tags %}
+                <div {{ component_vars.attrs }} >
+                </div>
+            """
+
+        template_str: types.django_html = """
+            {% load component_tags %}
+            {% component "test" attrs:@click.stop="dispatch('click_event')" attrs:x-data="{hello: 'world'}" attrs:class=class_var %}
+            {% endcomponent %}
+        """
+        template = Template(template_str)
+        rendered = template.render(Context({"class_var": "padding-top-8"}))
+        self.assertHTMLEqual(
+            rendered,
+            """
+            <div @click.stop="dispatch('click_event')" x-data="{hello: 'world'}" class="padding-top-8" >
+            </div>
+            """,
+        )
+
+    def test_attrs_accessible_in_get_context_data(self):
+        @component.register("test")
+        class AttrsComponent(component.Component):
+            template: types.django_html = """
+                {% load component_tags %}
+                <div {{ attrs }} >
+                </div>
+            """
+
+            def get_context_data(self, *args, attrs):
+                return {"attrs": attrs}
+
+        template_str: types.django_html = """
+            {% load component_tags %}
+            {% component "test" attrs:@click.stop="dispatch('click_event')" attrs:x-data="{hello: 'world'}" attrs:class=class_var %}
+            {% endcomponent %}
+        """
+        template = Template(template_str)
+        rendered = template.render(Context({"class_var": "padding-top-8"}))
+        self.assertHTMLEqual(
+            rendered,
+            """
+            <div @click.stop="dispatch('click_event')" x-data="{hello: 'world'}" class="padding-top-8" >
+            </div>
+            """,
+        )
+
+    def test_attrs_immutable(self):
+        @component.register("test")
+        class AttrsComponent(component.Component):
+            template: types.django_html = """
+                {% load component_tags %}
+                <div {{ attrs }} >
+                </div>
+            """
+
+            def get_context_data(self, *args, attrs):
+                attrs["my_super_key"] = "abc"
+                return {"attrs": attrs}
+
+        template_str: types.django_html = """
+            {% load component_tags %}
+            {% component "test" attrs:@click.stop="dispatch('click_event')" attrs:x-data="{hello: 'world'}" attrs:class=class_var %}
+            {% endcomponent %}
+        """
+        template = Template(template_str)
+        with self.assertRaisesMessage(TypeError, "cannot change object - object is immutable"):
+            template.render(Context({"class_var": "padding-top-8"}))
+
+    def test_attrs_tag(self):
+        @component.register("test")
+        class AttrsComponent(component.Component):
+            template: types.django_html = """
+                {% load component_tags %}
+                <div {% merge_attrs component_vars.attrs class+="added_class" data-id=123 %}>
```

Comment:
Also note here that I used component_vars.attrs to access the attributes. It's a bit wordy, it would've been awesome if it could be something simple like $attrs, but 1) dollar sign is not a valid identifier character, and 2) succint names can be achieved later. Now it's imo important that things work.